From 7d1842dcd0fd992f629cbc4860c812a3cff89ce6 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Tue, 21 Sep 2021 23:18:24 +0200
Subject: [PATCH 138/954] feat(stm32mp2): add helper to get DDRDBG base address

Add a function to get DDRDBG peripheral IO memory base address.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 265f21cd72aa35c6b741e76773a9b055e7a20dc6 feat(stm32mp2): add helper to get DDRDBG base address]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: I20d14fca49528c296c1f7d49a66129d932f44e49
---
 plat/st/stm32mp2/include/stm32mp2_private.h | 3 +++
 plat/st/stm32mp2/stm32mp2_private.c         | 5 +++++
 2 files changed, 8 insertions(+)

diff --git a/plat/st/stm32mp2/include/stm32mp2_private.h b/plat/st/stm32mp2/include/stm32mp2_private.h
index 4c9eae0f08..9d74fed101 100644
--- a/plat/st/stm32mp2/include/stm32mp2_private.h
+++ b/plat/st/stm32mp2/include/stm32mp2_private.h
@@ -30,4 +30,7 @@ uintptr_t stm32_risaf_get_memory_base(int instance);
 /* Get RISAF memory size in bytes from platform instance ID */
 size_t stm32_risaf_get_memory_size(int instance);
 
+/* Get DDRDBG peripheral IO memory base address */
+uintptr_t stm32_ddrdbg_get_base(void);
+
 #endif /* STM32MP2_PRIVATE_H */
diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index b086169bf1..ab4b20dc08 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -354,3 +354,8 @@ uintptr_t stm32_get_bkpr_boot_mode_addr(void)
 {
 	return tamp_bkpr(96);
 }
+
+uintptr_t stm32_ddrdbg_get_base(void)
+{
+	return DDRDBG_BASE;
+}
-- 
2.53.0

